### ISSUE TITLE
Behavior of 'logs' following 'up -d' and 'up'

### ISSUE THREAD

**Issue by iafilatov** [docker/compose#1486]
When you do `up -d`, the containers continue to work until their "natural death", i.e. are not killed if just one of them stops (like happens with `up` without `-d` - #741). But `logs` still detaches on each container's exit. I think `logs` should remain attached until no more output can be expected from the containers - until they *all* have exited. This should be a reasonable kind of behavior for simple`up` as well.

Also, if you do `logs` *after* all containers have stopped, it remains attached forever (or until you Ctrl+C out of it). This seems pointless as there's nothing to wait for at this point. Besides, you might want to pipe the logs to a pager to examine them, for example, but it won't work so well.

To sum it up, when compose is showing logs, and a container exits, it should check the status of other containers and remain attached as long it sees that any of them are still running.

To keep current behavior of `up`, the other containers must be stopped *before* the log printer exits, which seems trickier, but perhaps it could be possible to just teach the printer one more exit behavior: "exit if none running" in addition to current "exit on first stopped container". Then use the former for `logs` and the latter for `up`.

**Comment by dnephin** [comment]
I agree with the proposed `logs` behaviour. I think we should wait for all streams to exit when the containers are running, and exit immediately after printing the logs if the containers are stopped.

**Comment by kung-fu-tzu** [comment]
+1

**Comment by docteurklein** [comment]
It might worth to open an issue, but I'll try here before.

Another (related) feature would be to attach new containers as they come.
Imagine you `docker-compose logs` some containers (or all).
One service dies and is restarted (either by using `restart: always` or `restart` or `up service_a`.
I'd like docker-compose to watch docker events and detect that a new container with the same name is running, and thus streaming its output as well, without having to restart.

Thoughts?

**Comment by dnephin** [comment]
I agree logs probably should pick up new containers as they are created as well.

**Comment by aanand** [comment]
Yeah, I expressed the same thoughts here: https://github.com/docker/compose/issues/1610#issuecomment-117625334

**Comment by dnephin** [comment]
It looks like #1115 is a related issue about picking up new containers as they start

**Comment by aanand** [comment]
@iafilatov:

> I think `logs` should remain attached until no more output can be expected from the containers - until they all have exited. This should be a reasonable kind of behavior for simple `up` as well.

Implemented in #1754.

**Comment by dnephin** [comment]
I think this is covered by #1115 and #1866

**Comment by BeOleg** [comment]
@dnephin 
This might be unrelated, however it seems that when the logging driver cannot attach to the host, the whole stack fails to bootstrap.

Furthermore, if the stack is already running and the logging server hangs or disconnects, the stack fails and becomes unavailable.

`version: 2`

Error:
```
Removing nexchange_redis_1
Removing nexchange_postgis_1
Starting 28fe887ac395_28fe887ac395_nexchange_redis_1 ... 
Recreating 93808801a315_93808801a315_nexchange_postgis_1 ... 
Recreating 93808801a315_93808801a315_nexchange_postgis_1
Recreating 93808801a315_93808801a315_nexchange_postgis_1 ... error

ERROR: for 93808801a315_93808801a315_nexchange_postgis_1  Cannot start service postgis: failed to initialize logging driver: dial tcp: lookup tcp on 67.207.67.2:53: no such host

ERROR: for postgis  Cannot start service postgis: failed to initialize logging driver: dial tcp: lookup tcp on 67.207.67.2:53: no such host
ERROR: Encountered errors while bringing up the project.
```
